lib/pure/parseopt.nim

```diff
@@ -403,7 +426,8 @@ iterator getopt*(p: var OptParser): tuple[kind: CmdLineKind, key,
   ## Convenience iterator for iterating over the given
   ## `OptParser<#OptParser>`_.
   ##
-  ## There is no need to check for `cmdEnd` while iterating.
+  ## There is no need to check for `cmdEnd` while iterating. If using `getopt` 
+  ## with case switching, checking for `cmdEnd` is required.
```

Comment:
What is "case switching"? And why does the standard library docs now again assume that the reader never read about `case` statements which is a language construct? Here is what we should do: "Read the tutorials or at least skim them, then read the library documentation. You should do it this way for every programming language that you want to use."